persistence/pre_persistence/validator.go

```diff
@@ -0,0 +1,365 @@
+		return err
+	}
+	stakedTokens.Add(stakedTokens, stakedTokensToAddI)
+	// update values
+	val.ServiceURL = serviceURL
+	val.StakedTokens = BigIntToString(stakedTokens)
+	// marshal
+	bz, err := cdc.Marshal(val)
+	if err != nil {
+		return err
+	}
+	return db.Put(key, bz)
+}
+
+func (m *PrePersistenceContext) DeleteValidator(address []byte) error {
+	if exists, _ := m.GetValidatorExists(address); !exists {
+		return fmt.Errorf("does not exist in world state")
+	}
+	db := m.Store()
+	key := append(ValidatorPrefixKey, address...)
+	return db.Put(key, DeletedPrefixKey)
+}
+
+func (m *PrePersistenceContext) GetValidatorsReadyToUnstake(height int64, status int) (fishermen []*types.UnstakingActor, err error) {
+	db := m.Store()
+	unstakingKey := append(UnstakingValidatorPrefixKey, []byte(fmt.Sprintf("%d", height))...)
+	if has := db.Contains(unstakingKey); !has {
+		return nil, nil
+	}
+	val, err := db.Get(unstakingKey)
+	if err != nil {
+		return nil, err
+	}
+	if val == nil {
+		return make([]*types.UnstakingActor, 0), nil
+	}
+	unstakingActors := types.UnstakingActors{}
+	if err := proto.Unmarshal(val, &unstakingActors); err != nil {
+		return nil, err
+	}
+	for _, sn := range unstakingActors.UnstakingActors {
+		fishermen = append(fishermen, sn)
+	}
+	return
+}
+
+func (m *PrePersistenceContext) GetValidatorStatus(address []byte) (status int, err error) {
+	val, exists, err := m.GetValidator(address)
+	if err != nil {
+		return ZeroInt, err
+	}
+	if !exists {
+		return ZeroInt, fmt.Errorf("does not exist in world state")
+	}
+	return int(val.Status), nil
+}
+
+func (m *PrePersistenceContext) SetValidatorUnstakingHeightAndStatus(address []byte, unstakingHeight int64, status int) error {
+	validator, exists, err := m.GetValidator(address)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("does not exist in world state")
+	}
+	cdc := Cdc()
+	unstakingActors := types.UnstakingActors{}
+	db := m.Store()
+	key := append(ValidatorPrefixKey, address...)
+	validator.UnstakingHeight = unstakingHeight
+	validator.Status = int32(status)
+	// marshal
+	bz, err := cdc.Marshal(validator)
+	if err != nil {
+		return err
+	}
+	if err := db.Put(key, bz); err != nil {
+		return err
+	}
+	unstakingKey := append(UnstakingValidatorPrefixKey, []byte(fmt.Sprintf("%d", unstakingHeight))...)
+	if found := db.Contains(unstakingKey); found {
+		val, err := db.Get(unstakingKey)
+		if err != nil {
+			return err
+		}
+		if err := proto.Unmarshal(val, &unstakingActors); err != nil {
+			return err
+		}
+	}
+	unstakingActors.UnstakingActors = append(unstakingActors.UnstakingActors, &types.UnstakingActor{
+		Address:       validator.Address,
+		StakeAmount:   validator.StakedTokens,
+		OutputAddress: validator.Output,
+	})
+	unstakingBz, err := cdc.Marshal(&unstakingActors)
+	if err != nil {
+		return err
+	}
+	return db.Put(unstakingKey, unstakingBz)
+}
+
+func (m *PrePersistenceContext) GetValidatorPauseHeightIfExists(address []byte) (int64, error) {
```

Comment:
Likely, but in V0 we took the stance of keeping them separate so that we don't pigeonhole us into the actor specific functionality. To me, the current separation is the 'safer' approach right now, even though it's more code and not optimized